pages/en/lb4/Extending-LoopBack-4.md

```diff
@@ -0,0 +1,77 @@
+---
+lang: en
+title: 'Extending LoopBack 4'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Extending-LoopBack-4.html
+summary:
+---
+
+## Overview
+
+LoopBack 4 is designed to be highly extensible. For architectural rationale and motivation, see [Crafting LoopBack 4](Crafting-LoopBack-4.html). 
+
+## Building blocks for extensibility
+
+The `@loopback/context` module implements an IoC container (`Context`) as a service registry that supports [Dependency injection](Dependency-injection.html).
+
+![loopback-ioc](/images/lb4/loopback-ioc.png)
+
+The IoC container decouples service providers and consumers. A service provider can be bound to the context so that IT can be accessed by the binding key, which is the URI of the provider. Service consumers can then either locate the provider using the binding key or declare a dependency using `@inject('a-service-provider')` so that the service provider can be injected into the consumer class.
```

Comment:
> the binding key, which is the URI of the provider

I feel this sentence is misleading.

 - Our binding keys are not using [URI syntax](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax) as specified by RFC(s).

 - I don't think a binding key is an identifier of the provider, because a single provider can be bound to multiple binding keys.